enzyme/Enzyme/Enzyme.cpp

```diff
@@ -1120,11 +1120,19 @@ class EnzymeBase {
       return {};
     }
 
+#if LLVM_VERSION_MAJOR > 16
```

Comment:
Alternatively, we can just do return Options({....});

Reply:
Oh true, I like that better. Do you happen to know if the automatic conversion happens for both `std::optional` and `llvm::Optional`? Or do we still need the check?

Reply:
it should happen for both, we use this convention elsewhere